src/com/google/javascript/jscomp/Compiler.java

```diff
@@ -348,7 +348,9 @@ protected void reconcileOptionsWithGuards() {
     // But we do not want to see the warnings from OTI.
     if (options.getNewTypeInference()) {
       options.checkTypes = true;
-      options.setWarningLevel(DiagnosticGroups.OLD_CHECK_TYPES, CheckLevel.OFF);
+      if (!options.reportOldTypeInferenceErrorsUnderNTI) {
```

Comment:
I was going to suggest `reportOldTypeInferenceErrorsUnderNewTypeInference` but that's whole a lot of camelCase...